### PR TITLE
Create a plugin process function for each plugin type.

### PR DESCRIPTION
**Pull request by e0ipso**
And move documentation accordingly.

### CODE REVIEW

**Comment by e0ipso** [comment]
This one should be pretty straight forward.

**Comment by e0ipso** [comment]
@amitaibu ready for review.

File: restful.module [hunk 1]
```diff
@@ -16,6 +16,22 @@ function restful_ctools_plugin_directory($module, $plugin) {
...
  * Add defaults values to the restful related plugins.
+ */
+function restful_plugin_process(&$plugin, $info) {
+  // Common operations.
+  $plugin += array(
+    'description' => '',
+  );
+
+  // Call the plugin specific process functions.
+  $function = 'restful_plugin_process_' . $info['type'];
```
Comment (amitaibu) [inline restful.module:27]: since we have only 2 plugings that need a process, lets avoid function_exist() and just call those explicitly?

Comment (e0ipso) [inline restful.module:27]: We will have to check the type of the executing plug-in then.

Even if the function is empty it's the best place to document the options of the plug-in.

**Comment by amitaibu** [comment]
Thanks :)

**Pull request merged**